File: src/Sample/Plumix.Sample/Demos/Material/DrawerHeadersDemoPage.cs
using Avalonia.Media;
using Plumix.Material;
using Plumix.Rendering;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source: dart_sample/lib/demos/material/drawer_headers_demo_page.dart

public sealed class DrawerHeadersDemoPage : StatefulWidget
{
    public override State CreateState() => new DrawerHeadersDemoPageState();

    private sealed class DrawerHeadersDemoPageState : State
    {
        private bool _alternateDecoration;
        private int _detailsPressed;

        public override Widget Build(BuildContext context)
        {
            var plainColor = _alternateDecoration ? Colors.DarkSlateBlue : Colors.Indigo;
            return new Column(
                crossAxisAlignment: CrossAxisAlignment.Stretch,
                spacing: 10,
                children:
                [
                    new Text("DrawerHeader + UserAccountsDrawerHeader", fontSize: 20, color: Colors.Black),
                    new Text(
                        "Status-bar padding, animated decoration, account pictures, details toggle, and semantics.",
                        fontSize: 14,
                        color: Colors.DimGray),
                    new Row(
                        spacing: 8,
                        children:
                        [
                            new TextButton(
                                child: new Text(_alternateDecoration ? "Decoration B" : "Decoration A"),
                                onPressed: () => SetState(() => _alternateDecoration = !_alternateDecoration)),
                            new Text("details=$_detailsPressed", color: Colors.Black),
                        ]),
                    new Expanded(
                        child: new Row(
                            crossAxisAlignment: CrossAxisAlignment.Start,
                            spacing: 12,
                            children:
                            [
                                new Expanded(
                                    child: new DrawerHeader(
                                        decoration: new BoxDecoration(Color: plainColor),
                                        child: new Align(
                                            alignment: Alignment.BottomLeft,
                                            child: new Text("Plain header", color: Colors.White)))),
                                new Expanded(
                                    child: new UserAccountsDrawerHeader(
                                        decoration: new BoxDecoration(
                                            Color: _alternateDecoration ? Colors.Teal : Color.Parse("#FF673AB7")),
                                        accountName: new Text("Ada Lovelace"),
                                        accountEmail: new Text("ada@example.test"),
                                        currentAccountPicture: new CircleAvatar(child: new Text("AL")),
                                        otherAccountsPictures:
                                        [
                                            new CircleAvatar(radius: 20, child: new Text("GH")),
                                            new CircleAvatar(radius: 20, child: new Text("CS")),
                                        ],
                                        onDetailsPressed: () => SetState(() => _detailsPressed++))),
                            ])),
                ]);
        }
    }
}

File: dart_sample/lib/demos/material/drawer_headers_demo_page.dart
import 'package:flutter/material.dart';

class DrawerHeadersDemoPage extends StatefulWidget {
  const DrawerHeadersDemoPage({super.key});

  @override
  State<DrawerHeadersDemoPage> createState() => _DrawerHeadersDemoPageState();
}

class _DrawerHeadersDemoPageState extends State<DrawerHeadersDemoPage> {
  bool _alternateDecoration = false;
  int _detailsPressed = 0;

  @override
  Widget build(BuildContext context) {
    final Color plainColor = _alternateDecoration
        ? Colors.blueGrey
        : Colors.indigo;
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 10,
      children: <Widget>[
        const Text(
          'DrawerHeader + UserAccountsDrawerHeader',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Status-bar padding, animated decoration, account pictures, details toggle, and semantics.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Row(
          spacing: 8,
          children: <Widget>[
            TextButton(
              onPressed: () =>
                  setState(() => _alternateDecoration = !_alternateDecoration),
              child: Text(
                _alternateDecoration ? 'Decoration B' : 'Decoration A',
              ),
            ),
            Text('details=$_detailsPressed'),
          ],
        ),
        Expanded(
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            spacing: 12,
            children: <Widget>[
              Expanded(
                child: DrawerHeader(
                  decoration: BoxDecoration(color: plainColor),
                  child: const Align(
                    alignment: Alignment.bottomLeft,
                    child: Text(
                      'Plain header',
                      style: TextStyle(color: Colors.white),
                    ),
                  ),
                ),
              ),
              Expanded(
                child: UserAccountsDrawerHeader(
                  decoration: BoxDecoration(
                    color: _alternateDecoration
                        ? Colors.teal
                        : Colors.deepPurple,
                  ),
                  accountName: const Text('Ada Lovelace'),
                  accountEmail: const Text('ada@example.test'),
                  currentAccountPicture: const CircleAvatar(child: Text('AL')),
                  otherAccountsPictures: const <Widget>[
                    CircleAvatar(radius: 20, child: Text('GH')),
                    CircleAvatar(radius: 20, child: Text('CS')),
                  ],
                  onDetailsPressed: () => setState(() => _detailsPressed++),
                ),
              ),
            ],
          ),
        ),
      ],
    );
  }
}
